DEFEDIT-982 Fix issue with lingering, invalid, MenuItems.

ragnard-king opened the pull request:
We can now invalidate a specific menu based on :id instead of always
invalidating the entire menubar and re-creating it.

ragnard-king committed:
DEFEDIT-982 Fix issue with lingering, invalid, MenuItems.

We can now invalidate a specific menu based on :id instead of always
invalidating the entire menubar and re-creating it.

erikangelin-king reviewed (changes requested):
Also looks good, just a question about wiping the invalid-menubar-items

ragnard-king committed:
DEFEDIT-982 Review fixes

- remove redundant comp
- clear invalidated menubar items also when rebuilding entire menubar
